tools/fpgabist/bist_common.py

```diff
@@ -52,7 +51,8 @@ def get_all_fpga_bdfs():
 
 
 def get_bdf_from_args(args):
-    pattern = "(?P<bus>\w{2}):(?P<device>\d{2})\.(?P<function>\d).*?."
+    pattern = ('(?P<bus>[a-fA-F0-9]{2}):'
+               '(?P<device>[a-fA-F0-9]{2})\.(?P<function>[a-fA-F0-9]).*?.')
```

Comment:
Is this a duplicate from lines 39,40? If so we should probably make this a global variable at the top...